api/app/Policies/UserPolicy.php

```diff
@@ -118,4 +181,57 @@ protected function applicantHasAppliedToPoolInTeams(User $applicant, $teamIds)
             })
             ->exists();
     }
+
+    /*******************  ROLE CHECKING  *******************/
+
+    /**
+     * Function to check if the acting user can update a team based role
+     *
+     * @return bool
+     */
+    protected function teamAbleToCheck(User $actor, string $roleId, string $teamId)
+    {
+        $role = Role::findOrFail($roleId);
+        $team = Team::findOrFail($teamId);
+
+        switch ($role->name) {
+            case 'pool_operator':
+                return $actor->isAbleTo('assign-any-teamRole');
+            case 'process_operator':
+                return $actor->isAbleTo('update-any-processOperatorMembership') || $actor->isAbleTo('update-team-processOperatorMembership', $team);
+            case 'community_recruiter':
+                return $actor->isAbleTo('update-any-communityRecruiterMembership ') || $actor->isAbleTo('update-team-communityRecruiterMembership ', $team);
+            case 'community_admin':
+                return $actor->isAbleTo('update-any-communityAdminMembership ') || $actor->isAbleTo('update-team-communityAdminMembership ', $team);
+        }
+
+        return false; // reject unknown roles
+    }
+
+    /**
+     * Function to check if the acting user can update an individual role
+     *
+     * @return bool
+     */
+    protected function individualAbleToCheck(User $actor, string $roleId)
+    {
+        $role = Role::findOrFail($roleId);
+
+        switch ($role->name) {
+            case 'guest':
+                return $actor->isAbleTo('assign-any-role');
+            case 'base_user':
+                return $actor->isAbleTo('assign-any-role');
+            case 'applicant':
+                return $actor->isAbleTo('assign-any-role');
+            case 'request_responder':
+                return $actor->isAbleTo('assign-any-role');
+            case 'community_manager':
+                return $actor->isAbleTo('assign-any-role');
+            case 'platform_admin':
+                return $actor->isAbleTo('update-any-platformAdminMembership ');
```

Comment:
OR assign-any-role

Reply:
Yes! Added in https://github.com/GCTC-NTGC/gc-digital-talent/pull/10791/commits/fc1e51beadbc64041a1b461f41f67b04111de4c2